publisher-metadata/app/controllers/DatasetMetadataDCAT.java

```diff
@@ -0,0 +1,273 @@
+		 * The UUID for the landingspage URL is the metadata for this dataset contained in the xml.
+		 * The UUID is from dataset.metadata_file_identification
+		 *
+		 * Examples:
+		 * https://metadata.geopublisher.local/metadata/dataset/85b0f332-b73f-4207-8518-b88e3fa710e1.xml is dataset.metadata_file_identification
+		 * http://services.geopublisher.local/geoserver/Huurwoningen/wfs?service=wfs&version=2.0.0&request=GetFeature&typeName=Aantal_koop-_en_huurwoningen_per_buurt
+		 */ 
+
+		String baseServiceUrl = "http://services.geopublisher.local/"; // Waarschijnlijk uit een config file
+		String baseMetadataUrl = "http://metadata.geopublisher.local/"; // Waarschijnlijk uit een config file
+		// http://services.geodata-utrecht.nl
+
+		String geoserverUrl = "geoserver/";
+		String getFeatureURl = "/wfs?service=wfs&version=2.0.0&request=GetFeature&typeName=";
+
+		String metadataIdent = ds.get(dataset.metadataFileIdentification);
+
+		String typeName = ds.get(publishedServiceDataset2.layerName).replaceAll(" ", "_");
+		String nameSpace = ds.get(genericLayer.name).replaceAll(" ", "_");
+
+		SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
+
+		resultDataset.put("@type", "dcat:Dataset");
+		resultDataset.put("license", licentie);
+		resultDataset.put("landingspage", baseMetadataUrl+"metadata/dataset/"+metadataIdent+".xml");
+		resultDataset.put("accessLevel", ds.get(sourceDatasetVersion.confidential) ? "confidential" : "public");
+
+		// Distributions
+		// some info over the distributions formats
+		Map<String, String[]> distributionsTypes = new HashMap<>(); 
+		distributionsTypes.put("GML2", new String[] {"gml2", "text/xml; subtype=gml/2.1.2"});
+		distributionsTypes.put("GML3", new String[] {"gml3", "application/gml+xml; version=3.2"});
+		distributionsTypes.put("KML", new String[] {"kml", "application/vnd.google-earth.kml+xml"});
+		distributionsTypes.put("CSV", new String[] {"csv", "text/csv"});
+		distributionsTypes.put("SHAPE-ZIP", new String[] {"shape-zip", "application/zip"});
+		distributionsTypes.put("GeoJSON", new String[] {"application/json", "application/vnd.geo+json"});
+
+		for (String dt : distributionsTypes.keySet()) {
+			HashMap<String, String> distribution = new HashMap<>();
+			distribution.put("@type", "dcat:Distribution");
+			distribution.put("title", dt);
+			distribution.put("format", dt);
+			distribution.put("mediaType", distributionsTypes.get(dt)[1]);
+			distribution.put("downloadURL", baseServiceUrl+geoserverUrl+nameSpace+getFeatureURl+typeName+"&outputFormat="+distributionsTypes.get(dt)[0]);
+
+			distributions.add(distribution);
+		}
+		resultDataset.put("distribution", distributions);
+
+
+		// All info from XML
+		try {
+			MetadataDocument metadataDocument = mdf.parseDocument(ds.get(sourceDatasetMetadata.document));
+
+			try {
+				resultDataset.put("title", metadataDocument.getDatasetTitle());
+			} catch (NotFound nf) {
+				resultDataset.put("title",  null);

```

Comment:
doet niets...

Reply:
Zorgt ervoor dat het juist wel in de JSON terecht komt. Even nagaan wat technisch gewenst is.